src/manifest.ts

```diff
@@ -46,22 +47,40 @@ export async function writeManifest(manifest: Manifest) {
 
 /**
  * Returns true if the manifest is valid.
+ */
+export async function isValidManifest(): Promise<boolean> {
+  return await getManifest() != null;
+}
+
+/**
+ * Ensures the manifest is correct for running a function.
+ * The manifest must include:
+ * "executionApi": {
+ *   "access": "MYSELF"
+ * }
+ */
+export async function isValidRunManifest(): Promise<boolean> {
+  if (isValidManifest) {
+    const manifest = await getManifest();
+    if (manifest.executionApi && manifest.executionApi.access === 'MYSELF') {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * Reads manifest file from project root dir.
  * The manifest is valid if it:
  * - It exists in the project root.
  * - Is valid JSON.
  */
-export async function isValidManifest(): Promise<boolean> {
+// tslint:disable-next-line:no-any
+export async function getManifest(): Promise<any> {
```

Comment:
`isValidManifest` previously just checked if it was a valid JSON file, which is what `getValidJSON` does anyway. This returns a manifest if its valid.